### DISCUSSION TITLE
Postgres password authentication failed : docker self hosted

### DISCUSSION THREAD

**Discussion by armaneth95** [chatwoot/chatwoot#3323] [Self hosted] [answered]
Hi!
Am having trouble deploying Chatwoot in selfhosted mode on docker. I followed instructions from https://www.chatwoot.com/docs/self-hosted/deployment/docker

Below are the lines modified in **.env**

```
DEFAULT_LOCALE=en

REDIS_PASSWORD=password

POSTGRES_HOST=postgres

POSTGRES_DATABASE=chatwoot_production
POSTGRES_USERNAME=admin
POSTGRES_PASSWORD=password
RAILS_ENV=production

LOG_SIZE= 1024
```

I reported the previous Postgres database details in the **docker-compose.yaml** as recommended but when i run

`docker-compose run --rm rails bundle exec rails db:chatwoot_prepare`

it fails showing the errors below:

```
ActiveRecord::ConnectionNotEstablished: FATAL: password authentication failed for user "admin"
PG::ConnectionBad: FATAL: password authentication failed for user "admin"

```
It keeps failing like this even after removing all related containers, images and changing postgres username and password before giving a another try. What is awkward is that a test virtual machine, it works fine at the first try.

_Notice: Both my prod and test machines are running Ubuntu 20.04 with Docker CE_

Finally, the workaround I used was to connect to the postgres container and change the user password using psql command.

If anybody has an idea of what could be the problem, please let us know.

Thanks in advance

**Comment by vishnu-narayanan** [comment, marked as answer]
@armaneth95 Hey, could you please remove the data in `/data/postgres` and try again ? My guess is the data from first attempt is persisting since we are not using named volumes in production compose file. I assume you might have removed the volumes via `docker volumes` command.

**Reply by armaneth95** [reply]
Thanks for your answer.
It make sense and even more it works like a charm.

Merci!
